server/app/routes/utils.js

```diff
@@ -0,0 +1,170 @@
+'use strict';
+
+//exports
+module.exports = {
+  appPath,
+  dbPath,
+  bluebirdForTests,
+  createAdminUser,
+  createGuest,
+  createLocalUser,
+  credentials,
+  db,
+  describeIt,
+  ensureAuthenticated,
+  _err,
+  forceSyncDb,
+  getApp,
+  getModel,
+  getModels,
+  joinRootPath,
+  logInAgent,
+  logOutAgent,
+  responder: responder,
+  routerUse,
+  Sequelize,
+  should,
+  supertest,
+  syncDb
+}
+
+
+
+// dependencies
+const path = require( 'path' );
+const bluebird = bluebirdForTests();
+
+//declarations
+const appPath = joinRootPath( "./server/app" );
+const dbPath = joinRootPath( "./server/db" );
+global.paths = {routeUtils: joinRootPath( './server/app/routes/utils.js')}
+
+function addUser( userInfo ) {
+  return getModel('user')
+  .findOrCreate( userInfo )
+}
+
+function bluebirdForTests() {
+  const Bluebird = require( 'bluebird' );
+  return Bluebird;
+}
+
+function createAgent( ) {
+  return supertest().agent( getApp() )
+}
+
+function createAdminUser( userInfo ) {
+  userInfo.isAdmin = true;
+  return createLocalUser( userInfo )
+}
+
+function createGuest() {
+  return createAgent();
+}
+
+function createLocalUser( userInfo ) {
+  return bluebird.all( [addUser(userInfo), createAgent()])
+    .spread( (userInstance, agent) => loginAgent( agent, userInfo, userInstance ) )
+}
+
+function credentials( username, password, isAdmin, options ) {
+  return {
+    username,
+    password,
+    isAdmin
+  }
```

Comment:
@jdhang - I took options and isAdmin out. I added an "email" field because our user model requires an email currently. the complete login cycle now works like a charm. Also, you can use the "ensureAuthenticated" utility to ensure route authentication.